### ISSUE TITLE
stage5 error, makefile.darwin.clang.x86_64

### ISSUE THREAD

**Issue by jkleiser** [norayr/voc#1]
I just tried 'make -f makefile.darwin.clang.x86_64' on my Mac (OSX 10.11.2), and got "make: *** [stage5] Error 1". See attached file for details.
[stage5error.txt](https://github.com/norayr/voc/files/82881/stage5error.txt)

**Comment by norayr** [comment]
very interesting.

the problem is here:
In file included from SYSTEM.c:16:
src/lib/system/darwin/clang/x86_64/SYSTEM.h:14:10: fatal error: 'alloca.h' file not found
#include <alloca.h>

it seems that macosx machine on which I build and test voc, has this headers, while your machine - does not, for some reason.

I do not own that machine, and I did not install it. I only connect to it via ssh.
The owner told me that he has XCode installed.
Do you have any idea, why your system lacks /usr/include/alloca.h and how can it be installed on mac?

I'll make a research on this topic.

**Comment by norayr** [comment]
can you run this:

cd /usr/include
find . | grep alloca

find . | grep malloc

**Comment by jkleiser** [comment]
I will try that, probably later today. I am now working on another Mac, and I will report a problem in another issue (soon).

**Comment by jkleiser** [comment]
```
mac3:voc jkleiser$ cd /usr/include
-bash: cd: /usr/include: No such file or directory
```

**Comment by norayr** [comment]
seems that you have no development libraries installed.
sorry i was not able to find out how to install them on osx so far. (was busy with development). will update you.

also see my comment on the other issue you have opened.

**Comment by jkleiser** [comment]
I now found the reason for the missing /usr/include. It's a feature of "El Capitan", i.e. OSX 10.11. I had Xcode 7.2 installed, and I was able to build other tools based on LLVM (e.g. Pony), but I didn't have /usr/include (as I had on the other Mac with OSX 10.10). The secret command to use to get /usr/include was "xcode-select --install". See http://galvanist.com/post/130990812761/el-capitans-usrinclude.
Now I have built and installed voc successfully on El Capitan, and compiled and run MyHelloWorld.
